spec/requests/api/v1/merchants_request_spec.rb

```diff
@@ -173,5 +173,120 @@
       expect(response).to be_success
       expect(merchant_response["revenue"]).to eq "10.0"
     end
+
+    it 'returns favorite customer for merchant' do
+      merchant      = create(:merchant)
+      customer      = create(:customer)
+      customer2     = create(:customer)
+      invoice       = create(:invoice, merchant: merchant, customer: customer)
+      invoice2      = create(:invoice, merchant: merchant, customer: customer)
+      invoice3      = create(:invoice, merchant: merchant, customer: customer2)
+      transactions  = create(:transaction, invoice:invoice)
+      transactions2 = create(:transaction, invoice:invoice)
+      transactions3 = create(:transaction, invoice:invoice2)
+      transactions4 = create(:transaction, invoice:invoice3)
+
+      get "/api/v1/merchants/#{merchant.id}/favorite_customer"
+
+      favorite_customer_response = JSON.parse(response.body)
+
+      expect(response).to be_success
+      expect(favorite_customer_response["first_name"]).to eq(customer.first_name)
+      expect(favorite_customer_response["last_name"]).to eq(customer.last_name)
+    end
+
+    it 'returns top merchants when quantity 1 for merchant' do
+      merchant      = create(:merchant)
+      merchant2     = create(:merchant)
+      invoice       = create(:invoice, merchant: merchant)
+      invoice2      = create(:invoice, merchant: merchant2)
```

Comment:
We should try to condense these test setups down the line...if we have time

Reply:
I agree